Sources/PostgresNIO/New/PreparedStatement.swift

```diff
@@ -0,0 +1,40 @@
+/// A prepared statement.
+///
+/// Structs conforming to this protocol will need to provide the SQL statement to
+/// send to the server and a way of creating bindings are decoding the result.
+///
+/// As an example, consider this struct:
+/// ```swift
+/// struct Example: PreparedStatement {
+///     static var sql = "SELECT pid, datname FROM pg_stat_activity WHERE state = $1"
+///     typealias Row = (Int, String)
+///
+///     var state: String
+///
+///     func makeBindings() -> PostgresBindings {
+///         var bindings = PostgresBindings()
+///         bindings.append(.init(string: self.state))
+///         return bindings
+///     }
+///
+///     func decodeRow(_ row: PostgresNIO.PostgresRow) throws -> Row {
+///         try row.decode(Row.self)
+///     }
+/// }
+/// ```
+///
+/// Structs conforming to this protocol can then be used with `PostgresConnection.execute(_ preparedStatement:, logger:)`,
+/// which will take care of preparing the statement on the server side and executing it.
+public protocol PreparedStatement {
```

Comment:
we should require that this is Sendable :)

Reply:
Also we should prefix this one with `Postgres`. Reason: https://github.com/apple/swift-nio/blob/main/docs/public-api.md#1-no-global-namespace-additions-that-arent-prefixed